### PR TITLE
adding enhancement method for buildings to lua

### LINKED ISSUE #4931: Lua command to upgrade buildings
Request to extend the Lua command set to include a command to upgrade buildings.

### PR DESCRIPTION
**Pull request by MarkMcWire**

### CODE REVIEW

File: src/scripting/lua_map.cc [hunk 1]
```diff
@@ -5443,6 +5444,25 @@ int LuaBuilding::dismantle(lua_State* L) {
 	return 0;
 }
 
+/* RST
+   .. method:: enhance([keep_wares = false])
+
```
Comment (Noordfrees) [inline src/scripting/lua_map.cc:5449]:
```suggestion

      .. versionadded:: 1.1

```

I only just remembered that all new Lua functions should have this…

**Review by Noordfrees — COMMENTED** [review]
LGTM :) One comment inline, not tested yet

**Comment by MarkMcWire** [comment]
I changed the defaultai.cc, too. This line causes an exception in debug build, if you force the rangers building via AI Hint.
There is already a check in line 2839.

File: src/scripting/lua_map.cc [hunk 2]
```diff
@@ -5443,6 +5444,27 @@ int LuaBuilding::dismantle(lua_State* L) {
...
+      versionadded:: 1.1
```
Comment (Noordfrees) [inline src/scripting/lua_map.cc]: Almost. RST markup takes some getting used to…
```suggestion
      .. versionadded:: 1.1
```

**Comment by hessenfarmer** [comment]
> 
> 
> I changed the defaultai.cc, too. This line causes an exception in debug build, if you force the rangers building via AI Hint.
> There is already a check in line 2839.

Please do not change any things in a PR that are not related to the title of the PR, because this would make it impossible to have a complete changelog.    
For the behaviour described in the AI bug #4911 I am already working on the fix of it.

**Comment by hessenfarmer** [comment]
Alternatively you may open a new PR providing this as a hot fix with reference to the mentioned bug of course (use Re #XXXX syntax for this)

**Comment by MarkMcWire** [comment]
How can I undoing a commit? I've change the defaultai.cc in my repository, so it would be automatically contribute with every pull request.

**Comment by hessenfarmer** [comment]
> 
> 
> How can I undoing a commit? I've change the defaultai.cc in my repository, so it would be automatically contribute with every pull request.

Well, it is recommendable to not change things in master. Your master should be a copy of the upstream master (aka our main trunk) for your local changes that you need for playtesting developping you could then branch out a seperate branch (for example local_european_development or any other name you want) for providing pull request it is recomendabloe to branch out a fix branch. if it is related to an already known issue it is good practice to prefix it with the issue number but any name related to the content would do. (e.g. for this it could be lua_enhance_buildings. it is important not to have blanks in the branch name though. afterwards you could easily update your local europeans branch by pulling in upstream master, you could do the same with your master to have it consistent. and for every PR shortly before merging to ease merging and to be sure not to have merge conflicts.   

to solve the issue here you might branch out a new branch only containing the hot fix for the defaultai.cc (where the other changes need to be undone) and merge it first. 

@Noordfrees 
Alternatively we could allow this deviation here for one time as I am already working on fixing the complete issues which will be reflected in the changelog then anyway.

**Comment by Noordfrees** [comment]
The Refined GitHub browser extension adds a button to discard all changes to a file in a PR with one click, very convenient :) 

You can also use `git revert ae13bdf` to undo a specific commit

**Review by Noordfrees — APPROVED** [review]

**Pull request merged**